### PR TITLE
Menu update - improvements and mobile version

### PR DESCRIPTION
**Pull request by Tymek**
## About the changes

image

### CODE REVIEW

File: frontend/src/component/menu/Header/Header.tsx [hunk 1]
```diff
@@ -132,20 +135,22 @@ const Header: VFC = () => {
     const routes = getRoutes();
 
     const filterByMode = (route: INavigationMenuItem): boolean => {
```
Comment (kwasniew) [inline frontend/src/component/menu/Header/Header.tsx:137]: nested ifs with logic - I'd add a test

File: frontend/src/component/menu/Header/NavigationLink/NavigationLink.tsx [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import { ListItem, Link, styled } from '@mui/material';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { EnterpriseBadge } from 'component/common/EnterpriseBadge/EnterpriseBadge';
+import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
 import { Link as RouterLink } from 'react-router-dom';
 interface INavigationLinkProps {
     path: string;
     text: string;
     handleClose: () => void;
+    mode?: ('pro' | 'enterprise')[];
```
Comment (kwasniew) [inline frontend/src/component/menu/Header/NavigationLink/NavigationLink.tsx:10]: is it the only place we use this type? seems quite generic. Why is mode getting an array? Is it modes - plural?

File: frontend/src/component/menu/Header/NavigationMenu/NavigationMenu.tsx [hunk 3]
```diff
@@ -62,16 +79,13 @@ export const NavigationMenu = ({
...
-                            showUpdatedMenu &&
-                                options[i - 1]?.group &&
-                                options[i - 1]?.group !== option.group
-                        )}
-                        show={<Divider variant="middle" />}
-                        elseShow={null}
-                    />,
+                    Boolean(
+                        showUpdatedMenu &&
+                            options[i - 1]?.group &&
```
Comment (kwasniew) [inline frontend/src/component/menu/Header/NavigationMenu/NavigationMenu.tsx:84]: not very clear what those indexes are. I'd use some intention revealing variable/fn here

File: frontend/src/component/menu/Header/NavigationLink/NavigationLink.tsx [hunk 4]
```diff
@@ -52,6 +69,17 @@ const NavigationLink = ({ path, text, handleClose }: INavigationLinkProps) => {
             >
                 <StyledSpan />
                 {text}
+
+                <ConditionallyRender
+                    condition={Boolean(
+                        isPro() && props.mode?.includes('enterprise')
```
Comment (kwasniew) [inline frontend/src/component/menu/Header/NavigationLink/NavigationLink.tsx:75]: what is mode here? is it indicator if the feature is enterprise feature?

**Review by kwasniew — APPROVED** [review]

File: frontend/src/component/menu/Header/Header.tsx (hunk 1 above)
Comment (Tymek) [reply frontend/src/component/menu/Header/Header.tsx]: added

File: frontend/src/component/menu/Header/NavigationMenu/NavigationMenu.tsx (hunk 3 above)
Comment (Tymek) [reply frontend/src/component/menu/Header/NavigationMenu/NavigationMenu.tsx]: refactored to variable `previousGroup`

**Pull request merged**